pkg/kubectl/job.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type JobGeneratorV1 struct {
+	Name    string
+	Image   string
+	Command []string
+	From    *batchv1beta1.JobTemplateSpec
+}
+
+// Ensure it supports the generator pattern that uses parameter injection, for backwards compatibility.
+var _ Generator = &JobGeneratorV1{}
+
+// Ensure it supports the generator pattern that uses parameters specified during construction.
+var _ StructuredGenerator = &JobGeneratorV1{}
+
+func (JobGeneratorV1) ParamNames() []GeneratorParam {
+	return []GeneratorParam{
```

Comment:
if you're attaching this list to "create job", them I'm strongly against.  `kubectl create` commands are supposed to contain the minimum options.